### DISCUSSION TITLE
403 Forbidden after implement the FilamentUser contract

### DISCUSSION THREAD

**Discussion by MohcinBN** [filamentphp/filament#6998] [Help] [answered]
I recently began working with Filament. Initially, everything seemed fine after the installation. However, when I deployed my app, I encountered a 404 error page when attempting to access "/admin."

After several hours of searching, I discovered that I had overlooked the implementation of the FilamentUser contract. Upon rectifying this, I encountered a 403 Forbidden error on both the local and live app.

Could you please suggest a solution to this issue?

**Comment by pboivin** [comment]
What does your implementation of `canAccessFilament()` look like?

**Reply by MohcinBN** [reply]
`<?php

namespace App\Models;

// use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;

use Filament\Models\Contracts\FilamentUser;

class User extends Authenticatable implements FilamentUser
{
    use HasApiTokens, HasFactory, Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'name',
        'email',
        'password',
    ];

    /**
     * The attributes that should be hidden for serialization.
     *
     * @var array<int, string>
     */
    protected $hidden = [
        'password',
        'remember_token',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'email_verified_at' => 'datetime',
    ];

    public function canAccessFilament(): bool
    {
        return str_ends_with($this->email, '@mydomain.com');
    }
}
`

**Reply by pboivin** [reply]
If you return `true` there, are you still getting the 403? Are you using any auth plugins, custom policies, etc.?

**Reply by MohcinBN** [reply]
Thanks for your replies, If i returned true it work locally, but still give me 404 Not Found in production.
Yes i was using default Auth of laravel, it still there?!
*I renamed all Auth files to disabled it and also not working.

**Reply by danharrin** [reply]
read what you wrote in `canAccessFilament()` does that rule make sense for your application?

**Reply by MohcinBN** [reply]
```
    public function canAccessFilament(): bool
    {
        return str_ends_with($this->email, '@mydomain.com'); 
    }
```
I think yes, it make sense, and works perfectly in my local but in the production 404 Not Found!

**Reply by danharrin** [reply]
Okay well that doesnt make sense, not sure what we can do if you don't want to debug it 🤷‍♂️

**Reply by MohcinBN** [reply]
I'm trying with it.. don't worry :D

**Reply by MohcinBN** [reply, marked as answer]
Thank you all for your trying to help. 

It was my error, I'm missing to push my composer.json file for that contains the filament stuff.

I want to share in case someone else fall in the same situation
